get_velocity.py

```diff
@@ -0,0 +1,227 @@
+        return False
+    return True
+
+
+def get_week_representation(date_time) -> str:
+    year = date_time.year
+    week = date_time.isocalendar()[1]
+    week_str = f"{year}-{week:02d}"
+    return week_str
+
+
+def get_issue_value(
+    issue: GithubIssue,
+    labels: Optional[List[Tuple[str, int]]] = None,
+    value_without_label=0,
+):
+    if not labels:
+        return 1
+
+    issue_labels = [label.name for label in issue.labels]
+    value: Optional[int] = None
+    for label_name, label_value in labels:
+        if label_name in issue_labels:
+            value = label_value
+
+    if value is not None:
+        return value
+
+    click.echo(f"label not set: {issue.url}")
+    return value_without_label
+
+
+def get_closed_issues_per_week(
+    projects,
+    service,
+    interval=0,
+    labels: Optional[List[Tuple[str, int]]] = None,
+    value_without_label=0,
+    since: Optional[datetime] = None,
+):
+    since = since or (datetime.now() - timedelta(days=365 * 3))
+    week_numbers = {}
+    for week in get_week_numbers_since(since):
+        week_numbers[week] = 0
+    click.echo(week_numbers)
+    for project in projects:
+        namespace, project_name = project.split("/")
+        click.echo(f"{namespace}/{project_name}")
+        gh_project = service.get_project(namespace=namespace, repo=project_name)
+        for issue in get_closed_issues(gh_project, interval=interval):
+            if not is_counted(issue):
+                continue
+
+            week_str = get_week_representation(issue._raw_issue.closed_at)
+            value = get_issue_value(
+                issue, labels, value_without_label=value_without_label
+            )
+
+            if week_str in week_numbers:
+                week_numbers[week_str] += value
+
+    result = OrderedDict(sorted(week_numbers.items()))
+    for week, count in result.items():
+        click.echo(f"{week};{count}")
+    return week_numbers
+
+
+def get_week_numbers_since(since: datetime) -> List[str]:
+    delta = timedelta(days=7)
+
+    current_date = datetime.now()
+    week_numbers = []
+    while current_date > since:
+        week_numbers.append(get_week_representation(current_date))
+        current_date -= delta
+    return week_numbers
+
+
+DEFAULT_PROJECTS = [
+    "packit/ogr",
+    "packit/requre",
+    "packit/specfile",
+    "packit/packit",
+    "packit/research",
+    "packit/weekly-roles",
+    "packit/tokman",
+    "packit/wait-for-copr",
+    "packit/sandcastle",
+    "packit/packit-service-zuul",
+    "packit/packit-service-fedmsg",
+    "packit/packit-service",
+    "packit/dist-git-to-source-git",
+    "packit/deployment",
+    "packit/hardly",
+    "packit/dashboard",
+    "packit/packit.dev",
+    "packit/private",
+    "packit/packit-service-centosmsg",
+]
+
+
+@click.command(
+    help="Get the weekly number of issues closed",
+)
+@click.option(
+    "-t",
+    "--token",
+    type=click.STRING,
+    help="GitHub token",
+)
```

Comment:
env var would be nice 😇

Reply:
This should work already, try `TEAM_VELOCITY_TOKEN`..;)

```python
if __name__ == "__main__":
    velocity(auto_envvar_prefix="TEAM_VELOCITY")
```

Reply:
ha, missed that one